### DISCUSSION TITLE
AssertionError: The virtual device could not connect to ViGEmBus.

### DISCUSSION THREAD

**Discussion by ollieollieoxiclean** [trackmania-rl/tmrl#88] [Q&A] [answered]
I just finished the setup process for the installation and checking that it all works. Following the tutorial found here: https://github.com/trackmania-rl/tmrl/blob/master/readme/get_started.md
I've been trying to follow along with the, "If you want to test the pre-trained AI for raw screenshots" section. And after doing everything it tells me and putting in the command: python -m tmrl --test. It always does nothing and finishes with the error: AssertionError: The virtual device could not connect to ViGEmBus. I don't know much about how this works, so forgive me if I'm being dumb. I looked at some other people who had similar problems and they just started talking nonsense I couldn't understand so if someone could help me out that would be amazing. Thanks!

**Comment by yannbouteiller** [comment, marked as answer]
Hi, the error you encounter comes from the `vgamepad` library which apparently failed to install properly. Did pip prompt you with the vigembus installation?

**Reply by ollieollieoxiclean** [reply]
It did prompt me with the installation. I can try to uninstall and reinstall the gamepad library and try again. I'll see if that might work.

**Reply by ollieollieoxiclean** [reply]
That did work. Thank you!
